src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -118,22 +117,22 @@ end
 
 # Find X that is orthogonal, and B-orthogonal to Y, up to a tolerance tol.
 function ortho(X, Y, BY; tol=2eps(real(eltype(X))))
-    for i=1:size(X,2)
+    Threads.@threads for i=1:size(X,2)
         n = norm(@views X[:,i])
         X[:,i] ./= n
     end
 
     niter = 1
     ninners = zeros(Int,0)
     while true
-        BYX = BY'X
-        X .-= Y*BYX
+        BYX = blockdot(BY, X)
```

Comment:
Here and for every instance of blockdot/blockmul, can you add the actual formula (here `BY'X`) as a comment? That makes it easier to see what mathematical operation is done